failure in gdr_copy/uct_p2p_rma_test.put_short/0

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6079/label=vulcan-jenkins,worker=0/console
```
17:22:16 [       OK ] gdr_copy/uct_p2p_rma_test.get_zcopy/0 (0 ms)
17:22:16 [ RUN      ] gdr_copy/uct_p2p_rma_test.put_short/0
17:22:16 [     INFO ] memory_type:cuda 0 4.0g /scrap/jenkins/workspace/hpc-ucx-pr-3/label/vulcan-jenkins/worker/0/contrib/../test/gtest/uct/uct_test.cc:740: Failure
17:22:19 Value of: cerr == cudaSuccess
17:22:19   Actual: false
17:22:19 Expected: true
17:22:26 1x{0..4.0g} nocomp 64.0k 
17:22:26 
[  FAILED  ] gdr_copy/uct_p2p_rma_test.put_short/0, where GetParam() = gdr_copy/gdrcopy0 (10822 ms)
17:22:26 [----------] 3 tests from gdr_copy/uct_p2p_rma_test (10822 ms total)
...
17:35:25 [  PASSED  ] 2297 tests.
17:35:25 [  FAILED  ] 1 test, listed below:
17:35:25 [  FAILED  ] gdr_copy/uct_p2p_rma_test.put_short/0, where GetParam() = gdr_copy/gdrcopy0
17:35:25 
17:35:25  1 FAILED TEST
17:35:26 
make: *** [test] Error 1
17:35:26 make: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr-3/label/vulcan-jenkins/worker/0/build-test/test/gtest'
```

--- a/test/gtest/common/test_helpers.cc
+++ b/test/gtest/common/test_helpers.cc
@@ -134,6 +134,10 @@ uint16_t get_port() {
     return port;
 }
 
+void *mmap_fixed_address() {
+    return (void*)0xff0000000;
+}
+
 namespace detail {
 
 message_stream::message_stream(const std::string& title) {

--- a/test/gtest/common/test_helpers.h
+++ b/test/gtest/common/test_helpers.h
@@ -102,6 +102,12 @@ uint16_t get_port();
 
 
 /**
+ * Address to use for mmap(FIXED)
+ */
+void *mmap_fixed_address();
+
+
+/**
  * Return the IP address of the given interface address.
  */
 template <typename S>

--- a/test/gtest/ucp/test_ucp_fence.cc
+++ b/test/gtest/ucp/test_ucp_fence.cc
@@ -149,7 +149,7 @@ protected:
         params.length     = memheap_size;
         params.flags      = GetParam().variant;
         if (params.flags & UCP_MEM_MAP_FIXED) {
-            params.address  = (void *)(uintptr_t)0xFF000000;
+            params.address  = ucs::mmap_fixed_address();
             params.flags   |= UCP_MEM_MAP_ALLOCATE;
         } else {
             memheap = malloc(memheap_size);

--- a/test/gtest/ucp/test_ucp_memheap.cc
+++ b/test/gtest/ucp/test_ucp_memheap.cc
@@ -6,7 +6,10 @@
 */
 
 #include "test_ucp_memheap.h"
+
+#include <common/test_helpers.h>
 #include <ucs/sys/sys.h>
+
 
 std::vector<ucp_test_param>
 test_ucp_memheap::enum_test_params(const ucp_params_t& ctx_params,
@@ -59,7 +62,7 @@ void test_ucp_memheap::test_nonblocking_implicit_stream_xfer(nonblocking_send_fu
         params.address = memheap;
         params.flags   = params.flags & (~(UCP_MEM_MAP_ALLOCATE|UCP_MEM_MAP_FIXED));
     } else if (params.flags & UCP_MEM_MAP_FIXED) {
-        params.address = (void *)(uintptr_t)0xFF000000;
+        params.address = ucs::mmap_fixed_address();
     } else {
         params.address = NULL;
         params.flags  |= UCP_MEM_MAP_ALLOCATE;
@@ -169,7 +172,7 @@ void test_ucp_memheap::test_blocking_xfer(blocking_send_func_t send,
         params.address = memheap;
         params.flags   = params.flags & (~(UCP_MEM_MAP_ALLOCATE|UCP_MEM_MAP_FIXED));
     } else if (params.flags & UCP_MEM_MAP_FIXED) {
-        params.address = (void *)(uintptr_t)0xFF000000;
+        params.address = ucs::mmap_fixed_address();
         params.flags |= UCP_MEM_MAP_ALLOCATE;
     } else {
         params.address = NULL;

--- a/test/gtest/ucp/test_ucp_mmap.cc
+++ b/test/gtest/ucp/test_ucp_mmap.cc
@@ -338,8 +338,7 @@ UCS_TEST_P(test_ucp_mmap, fixed) {
 
     for (int i = 0; i < 1000 / ucs::test_time_multiplier(); ++i) {
         size_t size = (i + 1) * ((i % 2) ? 1000 : 1);
-        uintptr_t uptr = 0xFF000000;
-        void *ptr = (void *)uptr;
+        void *ptr = ucs::mmap_fixed_address();
 
         ucp_mem_h memh;
         ucp_mem_map_params_t params;

--- a/test/gtest/uct/test_mem.cc
+++ b/test/gtest/uct/test_mem.cc
@@ -114,7 +114,7 @@ UCS_TEST_P(test_mem, md_fixed) {
     const size_t            page_size   = ucs_get_page_size();
     const size_t            n_tryes     = 101;
     uct_alloc_method_t      meth;
-    uintptr_t               p_addr      = 0xFF000000;
+    void*                   p_addr      = ucs::mmap_fixed_address();
     size_t                  n_success;
 
     uct_allocated_memory_t  uct_mem;
@@ -141,24 +141,24 @@ UCS_TEST_P(test_mem, md_fixed) {
             for (j = 0; j < n_tryes; ++j) {
                 meth = UCT_ALLOC_METHOD_MD;
 
-                status = uct_mem_alloc((void *)p_addr, 1,
+                status = uct_mem_alloc(p_addr, 1,
                                        UCT_MD_MEM_FLAG_FIXED|
                                        UCT_MD_MEM_ACCESS_ALL,
                                        &meth, 1, &pd, 1, "test", &uct_mem);
                 if (status == UCS_OK) {
                     ++n_success;
                     EXPECT_EQ(meth, uct_mem.method);
-                    EXPECT_EQ(p_addr, (uintptr_t)uct_mem.address);
+                    EXPECT_EQ(p_addr, uct_mem.address);
                     EXPECT_GE(uct_mem.length, (size_t)1);
                     /* touch the page*/
                     memset(uct_mem.address, 'c', uct_mem.length);
-                    EXPECT_EQ(*(char *)p_addr, 'c');
+                    EXPECT_EQ(*(char*)p_addr, 'c');
                     status = uct_mem_free(&uct_mem);
                 } else {
                     EXPECT_EQ(status, UCS_ERR_NO_MEMORY);
                 }
 
-                p_addr += 2 * page_size;
+                p_addr = (char*)p_addr + (2 * page_size);
             }
 
             EXPECT_GT(n_success, (size_t)0);
@@ -177,7 +177,7 @@ UCS_TEST_P(test_mem, mmap_fixed) {
     const size_t            page_size   = ucs_get_page_size();
     const size_t            n_tryes     = 101;
     uct_alloc_method_t      meth;
-    uintptr_t               p_addr      = 0xFF000000;
+    void*                   p_addr      = ucs::mmap_fixed_address();
     size_t                  n_success;
 
     uct_allocated_memory_t  uct_mem;
@@ -188,22 +188,22 @@ UCS_TEST_P(test_mem, mmap_fixed) {
     for (i = 0; i < n_tryes; ++i) {
         meth = (i % 2) ? UCT_ALLOC_METHOD_MMAP : UCT_ALLOC_METHOD_HUGE;
 
-        status = uct_mem_alloc((void *)p_addr, 1,
+        status = uct_mem_alloc(p_addr, 1,
                                UCT_MD_MEM_FLAG_FIXED|UCT_MD_MEM_ACCESS_ALL,
                                &meth, 1, NULL, 0, "test", &uct_mem);
         if (status == UCS_OK) {
             ++n_success;
             EXPECT_EQ(meth, uct_mem.method);
-            EXPECT_EQ(p_addr, (uintptr_t)uct_mem.address);
+            EXPECT_EQ(p_addr, uct_mem.address);
             EXPECT_GE(uct_mem.length, (size_t)1);
             /* touch the page*/
             memset(uct_mem.address, 'c', uct_mem.length);
-            EXPECT_EQ(*(char *)p_addr, 'c');
+            EXPECT_EQ(*(char*)p_addr, 'c');
             status = uct_mem_free(&uct_mem);
         } else {
             EXPECT_EQ(status, UCS_ERR_NO_MEMORY);
         }
-        p_addr += 2 * page_size;
+        p_addr = (char*)p_addr + (2 * page_size);
     }
 }
 

